lib/src/postgrest_query_builder.dart

```diff
@@ -26,12 +26,26 @@ class PostgrestQueryBuilder extends PostgrestBuilder {
   /// Performs horizontal filtering with SELECT.
   ///
   /// ```dart
-  /// postgrest.from('users').select('id, messages');
+  /// postgrest.from('users').select<PostgrestList>('id, messages');
   /// ```
-  PostgrestFilterBuilder select([
+  ///
+  /// ```dart
+  /// postgrest.from('users').select<PostgrestListResponse>('id, messages', FetchOptions(count: CountOption.exact));
+  /// ```
+  /// By setting [FetchOptions.count] to non null or [FetchOptions.forceResponse] to `true`, the return type is `PostgrestResponse<T>`.
+  ///
+  /// Allowed types for [R] are:
+  /// - [PostgrestList]
+  /// - [PostgrestMap]
+  /// - [PostgrestMap?]
+  /// - [PostgrestListResponse]
+  /// - [PostgrestMapResponse]
+  /// - [PostgrestResponse]
+  PostgrestFilterBuilder<R> select<R>([
```

Comment:
I was going to suggest to extend `Object`, but this will break in a case user is doing this huh? I wonder if there is a better way of solving this issue.

```dart
final data = await supabase.from('table').select();
final firstRow = data.first; // The getter 'first' isn't defined for the type 'Object'.
```

```suggestion
  PostgrestFilterBuilder<R> select<R extends Object>([
```